Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -343,44 +351,127 @@ namespace Terrain
         AZ::Vector3 mainCameraPosition = mainView->GetCameraTransform().GetTranslation();
         CheckStacksForUpdate(mainCameraPosition);
 
-        for (auto& view : process.m_views)
+        // First pass - gather a list of all sectors that could render based on their status and lod, disregarding camera frustums.
+
+        const float minMovedDistanceSq = m_sampleSpacing * m_sampleSpacing; // This could probably be relaxed further, but is a good starting point.
+        if (m_previousCameraPosition.GetDistanceSq(mainCameraPosition) > minMovedDistanceSq)
         {
-            float minDistanceSq = 0.0f;
+            m_previousCameraPosition = mainCameraPosition;
```

Comment:
Suggestion: If you move the first pass and the second pass each into separate methods, each one will be shorter and easier to read through, and this first "if" statement could get flipped and just be an early return so that the entire thing doesn't have to be nested.

Reply:
Hmm... I think the case for pulling the first pass out to its own function is pretty strong, but I don't think the second pass deserves it. I think I'll update the comments to not explicitly refer to this as first/second pass - originally it was that, but now generating the candidate sectors doesn't happen every frame. That helps with the early return issue and makes `DrawMeshes()` itself more concerned about simple view-frustum culling. I think that'll make you 90% happy 😉